src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -102,14 +103,19 @@ InputStateMachineEngine::InputStateMachineEngine(std::unique_ptr<IInteractDispat
     THROW_HR_IF_NULL(E_INVALIDARG, _pDispatch.get());
 }
 
-bool InputStateMachineEngine::EncounteredWin32InputModeSequence() const noexcept
+void InputStateMachineEngine::WaitUntilDSR(DWORD timeout) const noexcept
 {
-    return _encounteredWin32InputModeSequence;
+    // Technically we should decrement the timeout with each iteration,
+    // but I suspect infinite spurious wake-ups are a theoretical problem.
+    while (!_lookingForDSR.load(std::memory_order::relaxed))

```

Comment:
Doesn't this condition need to go the other way? We want to `atomic_wait` while `_lookingForDSR` is `true`.

Reply:
I've done my brain in trying to figure this out

Reply:
Oh look @zadjii-msft! Dustin found the actual source of the bug. 😅
It's not that it timeouts, but rather that it doesn't wait for the answer to begin with.